Public/Src/Engine/Processes/SharedOpaqueOutputLogger.cs

```diff
@@ -0,0 +1,333 @@
+                    leaveOpen: false);
+            });
+
+            m_lazyWriteHeader = Lazy.Create(() =>
+            {
+                FileEnvelope.WriteHeader(m_lazyBxlWriter.Value.BaseStream, m_envelopeId);
+                return Unit.Void;
+            });
+        }
+
+        private const string SidebandFilePrefix = "Pip";
+        private const string SidebandFileSuffix = ".sideband";
+
+        /// <summary>
+        /// Given a root directory (<paramref name="searchRootDirectory"/>), returns the full path to the sideband file corresponding to process <paramref name="process"/>.
+        /// </summary>
+        public static string GetSidebandFileForProcess(PathTable pathTable, AbsolutePath searchRootDirectory, Process process)
+        {
+            Contract.Requires(searchRootDirectory.IsValid);
+
+            var semiStableHashX16 = string.Format(CultureInfo.InvariantCulture, "{0:X16}", process.SemiStableHash);
+            var subDirName = semiStableHashX16.Substring(0, 3);
+
+            return searchRootDirectory.Combine(pathTable, subDirName).Combine(pathTable, $"{SidebandFilePrefix}{semiStableHashX16}{SidebandFileSuffix}").ToString(pathTable);
+        }
+
+        /// <summary>
+        /// Finds and returns all sideband files that exist in directory denoted by <paramref name="directory"/>
+        /// </summary>
+        /// <remarks>
+        /// CODESYNC: must be consistent with <see cref="GetSidebandFileForProcess(PathTable, AbsolutePath, Process)"/>
+        /// </remarks>
+        public static string[] FindAllProcessPipSidebandFiles(string directory)
+        {
+            return Directory.Exists(directory)
+                ? Directory.EnumerateFiles(directory, $"{SidebandFilePrefix}*{SidebandFileSuffix}", SearchOption.AllDirectories).ToArray()
+                : CollectionUtilities.EmptyArray<string>();
+        }
+
+        /// <summary>
+        /// Creates a reader for the given sideband file.
+        /// </summary>
+        public static SharedOpaqueSidebandFileReader CreateSidebandFileReader(string filePath)
+        {
+            Contract.Requires(File.Exists(filePath));
+            return new SharedOpaqueSidebandFileReader(filePath);
+        }
+
+        /// <summary>
+        /// Returns all paths recorded in the <paramref name="filePath"/> file, even if the
+        /// file appears to be corrupted.
+        /// 
+        /// If file at <paramref name="filePath"/> does not exist, returns an empty iterator.
+        /// 
+        /// <seealso cref="SharedOpaqueSidebandFileReader.ReadRecordedPaths"/>
+        /// </summary>
+        public static string[] ReadRecordedPathsFromSidebandFile(string filePath)
+        {
+            if (!File.Exists(filePath))
+            {
+                return CollectionUtilities.EmptyArray<string>();
+            }
+
+            using (var reader = CreateSidebandFileReader(filePath))
+            {
+                reader.ReadHeader(ignoreChecksum: true);

```

Comment:
>ReadHeader [](start = 23, length = 10)

Considering this, I'd add an <exception > tag in the comments explaining that an exception will be thrown if the sideband file is corrupted

Reply:
This doesn't throw (because of `ignoreChecksum: true`).